### PR TITLE
Unknown option value 'no' for option 'clientFramework'

### LINKED ISSUE #19244: Unknown option value 'no' for option 'clientFramework'.
##### **Reproduce the error**

Unknown option value 'no' for option 'clientFramework'.

##### **Browsers and Operating System**

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #19244 

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by qmonmert** [comment]
application generated but:
image

**Review by atomfrede — DISMISSED** [review]

File: jdl/jhipster/default-application-options.js [hunk 1]
```diff
@@ -227,6 +234,14 @@ function getDefaultConfigForNewApplication(customOptions = {}) {
   if (!options[PACKAGE_FOLDER] && options[PACKAGE_NAME]) {
     options[PACKAGE_FOLDER] = options[PACKAGE_NAME].replace(/\./g, '/');
   }
+  let skipClient = options[SKIP_CLIENT];
+  if (skipClient === undefined) {
+    skipClient = options[CLIENT_FRAMEWORK] === undefined || options[CLIENT_FRAMEWORK] === NO;
```
Comment (mshima) [inline jdl/jhipster/default-application-options.js:239]: Need more time to check the behavior, but the default is angular, not skipClient.

Comment (qmonmert) [reply jdl/jhipster/default-application-options.js:239]: ok you told me, but it fixes the issue 🙂

Comment (qmonmert) [reply jdl/jhipster/default-application-options.js:239]: ok you told me, but it fixes the issue 🙂

File: jdl/jhipster/default-application-options.js [hunk 2]
```diff
@@ -227,6 +234,14 @@ function getDefaultConfigForNewApplication(customOptions = {}) {
...
+  }
+  if (skipClient) {
+    delete options[CLIENT_FRAMEWORK];
```
Comment (mshima) [inline jdl/jhipster/default-application-options.js:242]: Should not remove clientFramework.
Please set clientFramework instead (related to https://github.com/jhipster/generator-jhipster/blob/main/rfcs/5-jhipster-rfc-inclusive-property-in-generator.md)
```suggestion
    options[CLIENT_FRAMEWORK] = NO;
```
Related to https://github.com/jhipster/generator-jhipster/blob/main/rfcs/5-jhipster-rfc-inclusive-property-in-generator.md

**Review by mshima — COMMENTED** [review]
The generators should be fixed by using clientFrameworkNo with same behavior as skipClient.
`clientFrameworkNo || skipClient` instead of just `skipClient`.
The configuration files is becoming too complex.

**Comment by qmonmert** [comment]
@mshima with your suggestion, it breaks tests :/

**Comment by mshima** [comment]
I am on mobile now, there isn’t much I can do.
But need to add something like
```
dest.clientFrameworkNo = dest.clientFramework === CLIENT_FRAMEWORK_NO;
```
at
https://github.com/jhipster/generator-jhipster/blob/02a7cfc49926bd9de015bb550beabb14f727287c/generators/generator-base.js#L3029

maybe this too:
```
dest.skipClient = dest.skipClient || dest.clientFrameworkNo;
```

**Comment by qmonmert** [comment]
@mshima I revert your suggestion because I tried the issue with your suggestion and there is a bug (JDL stopped and asked questions)
image

**Review by mshima — APPROVED** [review]

**Comment by qmonmert** [comment]
@mshima I tested and it's ok 👍 we merge? we share the bounty?

**Comment by mshima** [comment]
@qmonmert please take the bounty

**Comment by qmonmert** [comment]
@mshima thanks

**Comment by qmonmert** [comment]
@jdubois @pascalgrimaud I claimed my bounty https://opencollective.com/generator-jhipster/expenses/89562 😉

**Comment by DanielFran** [comment]
@qmonmert approved

**Comment by qmonmert** [comment]
@DanielFran thanks 😉

**Pull request merged**